src/lsm/scan_lookup.zig

```diff
@@ -11,12 +11,14 @@ const GridType = @import("../vsr/grid.zig").GridType;
 const ScanType = @import("scan_builder.zig").ScanType;
 
 /// Implements the lookup logic for loading objects from scans.
-pub fn ScanLookupType(comptime Groove: type, comptime Storage: type) type {
+pub fn ScanLookupType(
+    comptime Groove: type,
+    comptime Scan: type,
+    comptime Storage: type,
+) type {
```

Comment:
We allow the `ScanLookup` to use indexes from one Groove, but perform the actual lookup in another one, since their `timestamp`s match.